File: multiple_3_or_5/multiple_3_or_5.py
def prog_sum(n: int) -> int:
    return n * (n + 1) / 2


def amount_of_division_in_range(number: int, n: int) -> int:
    return number // n - (1 if number % n == 0 else 0)


def solution(number):
    return (
        (
            3 * prog_sum(amount_of_division_in_range(number, 3))
            + 5 * prog_sum(amount_of_division_in_range(number, 5))
            - 15 * prog_sum(amount_of_division_in_range(number, 15))
        )
        if number > 0
        else 0
    )

File: multiple_3_or_5/multiple_3_or_5.java
public class Solution {
    private int prog_sum(int n) { return n * (n + 1) / 2; }

    private int amount_of_divisions_in_range(int number, int n) {
        return number / n - (number % n == 0 ? 1 : 0);
    }

    public int solution(int number) {
        return 3 * prog_sum(amount_of_divisions_in_range(number, 3)) +
            5 * prog_sum(amount_of_divisions_in_range(number, 5)) -
            15 * prog_sum(amount_of_divisions_in_range(number, 15));
    }
}